lib/pure/net.nim

```diff
@@ -511,6 +527,13 @@ proc connect*(socket: Socket, address: string, port = Port(0),
   
   when defined(ssl):
     if socket.isSSL:
+      try:
+        discard parseIpAddress(address)
+      except ValueError:
+        # Discard result in case OpenSSL version doesn't support SNI, or we're
+        # not using TLSv1+
+        discard SSL_set_tlsext_host_name(socket.sslHandle, address)
```

Comment:
I can't think of one. It's against the RFC to have IP addresses in the server name (RFC 3546, section 3.1):

    Literal IPv4 and IPv6 addresses are not permitted in "HostName".

Happy for suggestions however! :)